erpnext/assets/doctype/asset_category/asset_category.py

```diff
@@ -11,12 +11,29 @@
 class AssetCategory(Document):
 	def validate(self):
 		self.validate_finance_books()
+		self.validate_accounts()
 
 	def validate_finance_books(self):
 		for d in self.finance_books:
 			for field in ("Total Number of Depreciations", "Frequency of Depreciation"):
 				if cint(d.get(frappe.scrub(field)))<1:
 					frappe.throw(_("Row {0}: {1} must be greater than 0").format(d.idx, field), frappe.MandatoryError)
+	
+	def validate_accounts(self):
+		account_type_map = {
+			'fixed_asset_account': { 'account_type': 'Fixed Asset' },
+			'accumulated_depreciation_account': { 'account_type': 'Accumulated Depreciation' },
+			'depreciation_expense_account': { 'account_type': 'Expense' },
+			'capital_work_in_progress_account': { 'account_type': 'Capital Work in Progress' }
+		}
+		for d in self.accounts:
+			for account in account_type_map.keys():
+				if d.get(account):
+					account_type = frappe.db.get_value('Account', d.get(account), 'account_type')
+					if account_type != account_type_map[account]['account_type']:
+						frappe.throw(_("Row {}: {} should be a {} account".format(d.idx, frappe.bold(frappe.unscrub(account)),
```

Comment:
The error message doesn't sounds right, For Eg: If account type of fixed asset account is incorrect then the error message will be shown as "Row 1: **Fixed Asset Account** should be a **Fixed Asset** account".  The user won't understand the mistake. Account Type should be explicitly mentioned like "Account type for {} should be {}"

Reply:
Great Review! Thanks.